packages/block-library/src/social-link/edit.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { URLPopover } from '@wordpress/block-editor';
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	IconButton,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getIconBySite } from './social-list';
+
+const SocialLinkEdit = ( { attributes, setAttributes, isSelected } ) => {
+	const { url, site } = attributes;
+	const [ showURLPopover, setPopover ] = useState( false );
+	const classes = classNames(
+		'wp-social-link',
+		'wp-social-link-' + site,
+		{ 'wp-social-link__is-incomplete': ! url },
+	);
+
+	// Import icon.
+	const IconComponent = getIconBySite( site );
+
+	return (
+		<Button
+			className={ classes }
+			onClick={ () => setPopover( true ) }
+		>
+			<IconComponent />
+			{ isSelected && showURLPopover && (
+				<URLPopover
+					onClose={ () => setPopover( false ) }
+				>
+					<form
+						className="block-editor-url-popover__link-editor"
+						onSubmit={ ( event ) => {
+							event.preventDefault();
+							setPopover( false );
+						} } >
+						<div className="editor-url-input block-editor-url-input">
+							<input type="text"
+								value={ url }
+								onChange={ ( event ) => setAttributes( { url: event.target.value } ) }
+								placeholder={ __( 'Enter Address or Profile' ) }
```

Comment:
I was testing this: 

![2019-09-06-11-06-42-address-or-profile](https://user-images.githubusercontent.com/583546/64418308-0a21b400-d09b-11e9-8f31-13ee69ed595f.png)

and my first thought when seeing the placeholder text was: "_ok, I can add the full URL (twitter.com/nosolosw) or my nickname(nosolosw)_". It turns out the nickname approach wasn't possible. So, how about simplifying this text to:

```suggestion
								placeholder={ __( 'Enter Address' ) }
```